src/main/java/frc/robot/subsystems/elevator/ElevatorIOSim.java

```diff
@@ -1,15 +1,58 @@
 package frc.robot.subsystems.elevator;
 
+import static edu.wpi.first.units.Units.Meters;
+
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.MotionMagicVoltage;
+import edu.wpi.first.math.controller.ProfiledPIDController;
 import edu.wpi.first.units.Distance;
 import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.wpilibj.Timer;
+import lib.motors.TalonFXSim;
 
 public class ElevatorIOSim implements ElevatorIO {
+    private final TalonFXSim motor;
+
+    private final MotionMagicVoltage positionRequest = new MotionMagicVoltage(0);
+    private final DutyCycleOut powerRequest = new DutyCycleOut(0);
+
+    public ElevatorIOSim() {
+        motor =
+                new TalonFXSim(
+                        2,
+                        ElevatorConstants.GEAR_RATIO,
+                        0.000_01,
+                        ElevatorConstants.GEAR_RATIO
+                                * (2 * Math.PI * ElevatorConstants.DRUM_RADIUS));
+
+        motor.setProfiledController(
+                new ProfiledPIDController(
+                        ElevatorConstants.KP.get(),
+                        ElevatorConstants.KI.get(),
+                        ElevatorConstants.KD.get(),
+                        ElevatorConstants.TRAPEZOID_PROFILE));
+    }
+
     @Override
-    public void setPower(double power) {}
+    public void setPower(double power) {
+        powerRequest.withOutput(power);
+        motor.setControl(powerRequest);
+    }
 
     @Override
-    public void setHeight(MutableMeasure<Distance> height) {}
+    public void setHeight(MutableMeasure<Distance> height) {
+        inputs.heightSetpoint = height;
```

Comment:
Why do this if it's done in the `Elevator` class?

Reply:
07e672eb7f4940958e07732c1ae6947ca6e4369a